functime/cross_validation.py

```diff
@@ -21,18 +21,33 @@ def train_test_split(
     splitter : Callable[pl.LazyFrame, Tuple[pl.LazyFrame, pl.LazyFrame]]
         Function that takes a panel LazyFrame and returns tuple of train / test LazyFrames.
     """
+    if isinstance(test_size, float):
+        if test_size < 0 or test_size > 1:
+            raise ValueError("`test_size` must be between 0 and 1")
+    elif isinstance(test_size, int):
+        if test_size < 0:
+            raise ValueError("`test_size` must be greater than 0")
+    else:
+        raise TypeError("`test_size` must be int or float")
```

Comment:
Do you want/care about this kind of param validation?

Reply:
definitely. I was also interested in covering the case where `test_size` is bigger than the smallest series in the dataframe. This would force a `.collect` if using a lazyframe, though.

Reply:
We can add something as the following to achieve it: 
```py
...
def split(X):
    entity_col = X.columns[0]
    max_size = X.lazy().group_by(entity_col).agg(pl.count()).select(pl.min("count")).collect().item()
    if isinstance(test_size, int) and test_size > max_size:
        raise ValueError
    ...
...
```

The fact that `train_test_split` returns a callable requires to post-pone this check for when the dataframe is provided

Reply:
I like the fact we return a callable. It makes sense we postpone the checks when the actual data is passed. We could still perform some validation in the splitter definition (e.g. if <0).